js/modules/k6/experimental/tracing/propagator.go

```diff
@@ -9,6 +9,22 @@ type Propagator interface {
 	Propagate(traceID string) (http.Header, error)
 }
 
+// SamplerPropagator is an interface for trace context propagation
+// with support for sampling.
+type SamplerPropagator interface {
+	// Propagator is the base interface for trace context propagation
+	Propagator
+
+	// Sampler ensures the implementer is able to perform sampling decisions.
+	Sampler
+
+	// SamplingFlag should return the sampling flag to use
+	// as part of the trace context header, according to
+	// the sampler's decision of whether or not a given
+	// trace should be sampled.
+	SamplingFlag() string
+}
```

Comment:
I feel like we won't be having any *other* kind of propagator so maybe just have one interface ?

Reply:
As mentioned in another comment, I dropped the interface indeed.